controller/kubernetes_pod_controller.go

```diff
@@ -0,0 +1,273 @@
+		},
+	})
+
+	return kc
+}
+
+func (kc *KubernetesPodController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer kc.queue.ShutDown()
+
+	logrus.Infof("Start " + controllerAgentName)
+	if !controller.WaitForCacheSync(controllerAgentName, stopCh, kc.pStoreSynced, kc.pvStoreSynced, kc.pvcStoreSynced) {
+		return
+	}
+	for i := 0; i < workers; i++ {
+		go wait.Until(kc.worker, time.Second, stopCh)
+	}
+	<-stopCh
+	logrus.Infof("Shutting down" + controllerAgentName)
+}
+
+func (kc *KubernetesPodController) worker() {
+	for kc.processNextWorkItem() {
+	}
+}
+
+func (kc *KubernetesPodController) processNextWorkItem() bool {
+	key, quit := kc.queue.Get()
+	if quit {
+		return false
+	}
+	defer kc.queue.Done(key)
+	err := kc.syncHandler(key.(string))
+	kc.handleErr(err, key)
+	return true
+}
+
+func (kc *KubernetesPodController) handleErr(err error, key interface{}) {
+	if err == nil {
+		kc.queue.Forget(key)
+		return
+	}
+
+	if kc.queue.NumRequeues(key) < maxRetries {
+		logrus.Warnf("Error syncing Longhorn kubernetes pod %v: %v", key, err)
+		kc.queue.AddRateLimited(key)
+		return
+	}
+
+	utilruntime.HandleError(err)
+	logrus.Warnf("Dropping Longhorn kubernetes pod %v out of the queue: %v", key, err)
+	kc.queue.Forget(key)
+}
+
+func (kc *KubernetesPodController) syncHandler(key string) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, controllerAgentName+": fail to sync %v", key)
+	}()
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	pod, err := kc.pLister.Pods(namespace).Get(name)
+	if err != nil {
+		if datastore.ErrorIsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, "Error getting Pod: %s", name)
+	}
+	nodeID := pod.Spec.NodeName
+	if err := kc.forceDeletePod(pod, nodeID, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Force delete a pod when all of the below conditions are meet:
+// 1. StatefulSetPodDeletePolicy is StatefulSetPodDeletePolicyForceDelete
+// 2. pod is belonging to a StatefulSet
+// 3. node containing the pod is downed
+// 4. the pod is terminating
+// 5. pod has a PV with provisioner driver.longhorn.io
+func (kc *KubernetesPodController) forceDeletePod(pod *v1.Pod, nodeID string, namespace string) error {
+	deletionPolicy := types.StatefulSetPodDeletePolicyNever
+	if deletionSetting, err := kc.ds.GetSettingValueExisted(types.SettingNameStatefulSetPodDeletePolicy); err == nil {
+		deletionPolicy = types.StatefulSetPodDeletePolicy(deletionSetting)
+	}
+
+	if deletionPolicy == types.StatefulSetPodDeletePolicyNever {
+		return nil
+	}
+
+	if !isOwnedByStatefulSet(pod) {
+		return nil
+	}
+
+	isNodeDowned, err := kc.ds.IsNodeDownOrDeleted(nodeID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to evaluate Node %v for pod %v in forceDeletePod", nodeID, pod.Name)
+	}
+	if !isNodeDowned {
+		return nil
+	}
+
+	if pod.DeletionTimestamp == nil || !pod.DeletionTimestamp.After(time.Now()) {
```

Comment:
Have a look at the description of the [After](https://golang.org/pkg/time/#Time.After) function
It looks like you have it the wrong way around :)

What you effectively have written is:
```
return nil while delTime < now, 
it's best to think of these as values on a timeline.
so by writing delTime.After(now) your saying the deletionTime is still to come, but then you negate that condition.
Which means in your case the deletionTime has already (passed / happened in the past), which is the condition you would want to have for pod deletion.
```

Reply:
My bad. Yeah, I missed understand the `After()` function =]]] . Thank you so much

Reply:
not fix yet